### PR TITLE
Add please-upgrade-node

### PR DESCRIPTION
**Pull request by thorn0**
https://github.com/typicode/please-upgrade-node#readme

- [x] I’ve added tests to confirm my change works.
- (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Is this really a good idea? There should be some `SyntaxError` on old Node.js version, then `pleaseUpgradeNode()` can't run.

**Comment by thorn0** [comment]
You're right, but at least it catches Node 8.

File: tests_integration/__tests__/early-exit.js [hunk 1]
```diff
@@ -85,3 +85,19 @@ describe("throw error and show usage with something unexpected", () => {
...
+      writable: false,
+    });
+    runPrettier("cli", ["--help"]).test({ status: 1 });
+  } finally {
+    Object.defineProperty(process, "version", {
+      value: originalProcessVersion,
+      writable: false,
+    });
+  }
+});
```
Comment (evilebottnawi) [inline tests_integration/__tests__/early-exit.js]: I think we need test exit code here too

Comment (thorn0) [reply tests_integration/__tests__/early-exit.js]: That's what `test({ status: 1 });` does.

**Comment by fisker** [comment]
@thorn0 Add a entry file, make it ES5 or add post-install script?

File: tests_integration/__tests__/early-exit.js (hunk 1 above)
Comment (evilebottnawi) [reply tests_integration/__tests__/early-exit.js]: I should be a little more attentively sometimes, sorry

**Comment by thorn0** [comment]
@fisker Ideally, yes, but I think it's good enough as is. At least for now.

**Comment by fisker** [comment]
Let's see if people complains.

**Comment by GrahamCampbell** [comment]
This is causing false positives when prettier is packed with https://github.com/zeit/pkg. I'm unable to use prettier with node 12.16.1 after this change.

**Comment by fisker** [comment]
@GrahamCampbell I don't know if your problem still exists, but you can use `cli.js` as an entry now.

**Pull request merged**